stream-chat-android/src/main/java/com/getstream/sdk/chat/ChatUX.kt

```diff
@@ -0,0 +1,104 @@
+package com.getstream.sdk.chat
+
+import com.getstream.sdk.chat.navigation.ChatNavigationHandler
+import com.getstream.sdk.chat.navigation.ChatNavigator
+import com.getstream.sdk.chat.navigation.ChatNavigatorImpl
+import com.getstream.sdk.chat.style.ChatFonts
+import com.getstream.sdk.chat.style.ChatFontsImpl
+import com.getstream.sdk.chat.style.ChatStyle
+import com.getstream.sdk.chat.utils.strings.ChatStrings
+import com.getstream.sdk.chat.utils.strings.ChatStringsImpl
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.livedata.ChatDomain
+
+/**
+ * ChatUX handles any configuration for the Chat UI elements. It replaces the older Chat class.
+ *
+ * @param client the low level chat client
+ * @param chatDomain the chat domain interface used for offline storage and state
+ * @param fonts allows you to overwrite fonts
+ * @param strings allows you to customize strings
+ * @param navigationHandler navigation handler for customizing things such as the media browsing experience
+ * @param markdown interface to to customize the markdown parsing behaviour, useful if you want to use more markdown modules
+ * @param urlSigner url signing logic, enables you to add authorization tokens for images, video etc
+ *
+ * @see ChatMarkdown
+ * @see UrlSigner
+ * @see ChatStrings
+ * @see ChatFonts
+ */
+public class ChatUX internal constructor(
+    public val client: ChatClient,
+    public val chatDomain: ChatDomain,
+    public val fonts: ChatFonts,
+    public val strings: ChatStrings,
+    public val navigationHandler: ChatNavigationHandler? = null,
+    public val markdown: ChatMarkdown,
+    public val urlSigner: UrlSigner
+) {
+    public val version: String
+        get() = BuildConfig.BUILD_TYPE + ":" + BuildConfig.VERSION_NAME
+
+    public val navigator: ChatNavigator = ChatNavigatorImpl()
+
+    // TODO: would be nice to support configuring the loading icon in 1 place
```

Comment:
Would be nice to be able to set a default loading view here that's picked up by the other views. What do you think @samiuelson ?

Reply:
We are discussing it in the team on how to refactor `BaseStyle` code. We should have the possibility to customize global/common appearance and behaviour of our UI components. 

However, I'm sceptical about the idea of providing a global loading view setting. That's for a couple of reasons.
1. It would need to be passed as a factory or producer function instead of `View` instance to avoid a memory leak. That would overcomplicate the API.
2. I'm not convinced I'd need something like this as a developer. Loading indicators seems to be a thing which even if kept consistent across different places can be of a different size, variant, etc.